nomenclate/core/nomenclature.py

```diff
@@ -2,146 +2,295 @@
+            self._create_token_attr(token, value)
+            self.signal.fire()
+        else:
+            for token_attr in token_attrs:
+                if token == token_attr.token:
+                    setattr(self, token_attr.token, value)
+
+    def get_token_attr(self, token):
+        token_attr = getattr(self, token.lower())
+        if token_attr is None:
+            raise exceptions.SourceError('This nomenclate instance has no %s attribute set.' % token)
+        else:
+            return token_attr
+
+    def get_token_attrs(self):
+        return [value for name, value in iteritems(self) if isinstance(value, TokenAttr)]
+
+    def _create_token_attr(self, token, value):
+        self[token.lower()] = TokenAttr(label=value, token=token)
+
+    @staticmethod
+    def _validate_name_in_format_order(name, format_order):
+        """ For quick checking if a key token is part of the format order
+        """
+        if name not in [token.lower() for token in format_order]:
+            raise exceptions.FormatError('The name token %s is not found in the current format ordering' %
+                                         format_order)
+
+    def __eq__(self, other):
+        print ('comparing token dicts self then other\n\t%s\n\t%s' % (self, other))
+        if isinstance(other, self.__class__):
+            equal = False
+            self_attrs = self.get_token_attrs()
+            other_attrs = other.get_token_attrs()
+            if len(self_attrs) > len(other_attrs):
+                return False
+
+            for attr in self_attrs:
+                for other_attr in other_attrs:
+                    print('comparing %s to %s == %s' % (attr, other_attr, attr == other_attr))
+                    equal = True if attr == other_attr else False
+                    if equal:
+                        break
+            return equal
+        return False
+
+    def __getattr__(self, name):
+        try:
+            return self[name]
+        except KeyError:
+            object.__getattribute__(self, name)
+
+    def __repr__(self):
+        return ' '.join(['%s:%s' % (token_attr.token, token_attr.label) for token_attr in self.get_token_attrs()])
 
 
 class Nomenclate(object):
     """This class deals with renaming of objects in an approved pattern
     """
-    NAMING_FORMAT_HEADER_PATH = ['naming_formats']
-    DEFAULT_FORMAT_PATH = NAMING_FORMAT_HEADER_PATH + ['node', 'default']
-    def __init__(self, **kwargs):
-        """ Set default a
-        """
+    CONFIG_PATH = ['overall_config']
+
+    SUFFIXES_FORMAT_PATH = ['suffixes']
+    NAMING_FORMAT_PATH = ['naming_formats']
+    DEFAULT_FORMAT_PATH = NAMING_FORMAT_PATH + ['node', 'default']
+
+    OPTIONS_PATH = ['options']
+    VARIATION_PATH = OPTIONS_PATH + ['var']
+    SIDE_PATH = OPTIONS_PATH + ['side']
+
+    FORMAT_STRING_REGEX = r'([A-Za-z][^A-Z_.]*)'
+
+    def __init__(self, *args, **kwargs):
+        self.signal = Signal()
```

Comment:
I like that it is more obviously a signal handler from this naming, but maybe update_signal? or outdated_signal?